be/src/exec/es/es_scroll_parser.h

```diff
@@ -33,39 +33,59 @@ class ScrollParser {
 
     void set_params(const TupleDescriptor* descs, const std::map<std::string, std::string>* docvalue_context);
 
+    // This method is used to test private function _append_value_from_json_val
+    Status test_append_value_from_json_val(Column* column, const TypeDescriptor& type_desc, const rapidjson::Value& col,
+                                           bool pure_doc_value);
+
 private:
-    static bool _pure_doc_value(const rapidjson::Value& obj);
+    static bool _is_pure_doc_value(const rapidjson::Value& obj);
 
     template <PrimitiveType type, class CppType = RunTimeCppType<type>>
     static void _append_data(Column* column, CppType& value);
 
     static void _append_null(Column* column);
 
-    Status _append_value_from_json_val(Column* column, PrimitiveType type, const rapidjson::Value& col,
+    Status _append_value_from_json_val(Column* column, const TypeDescriptor& type_desc, const rapidjson::Value& col,
                                        bool pure_doc_value);
 
     Slice _json_val_to_slice(const rapidjson::Value& val);
 
     template <PrimitiveType type, typename T = RunTimeCppType<type>>
-    Status _append_int_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
+    static Status _append_int_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
 
     template <PrimitiveType type, typename T = RunTimeCppType<type>>
-    Status _append_float_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
+    static Status _append_float_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
 
-    Status _append_bool_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
+    static Status _append_bool_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
 
     template <PrimitiveType type, typename T = RunTimeCppType<type>>
-    Status _append_date_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
+    static Status _append_date_val(const rapidjson::Value& col, Column* column, bool pure_doc_value);
+
+    // In ES, array maybe stored as two different format, we should deal with it separately.
```

Comment:
The representation of array types in `_source` and `doc_value` (column storage) is inconsistent in Elasticsearch
, we need to do different processing to show consistent behavior